sqrl-openai/src/main/java/com/datasqrl/openai/completions.java

```diff
@@ -1,19 +1,40 @@
 package com.datasqrl.openai;
 
+import com.datasqrl.openai.util.P99LatencyTracker;
 import com.google.auto.service.AutoService;
+import org.apache.flink.metrics.Counter;
+import org.apache.flink.metrics.Gauge;
 import org.apache.flink.table.functions.FunctionContext;
 import org.apache.flink.table.functions.ScalarFunction;
 
+import java.util.concurrent.TimeUnit;
+
 import static com.datasqrl.openai.RetryUtil.executeWithRetry;
 
 @AutoService(ScalarFunction.class)
 public class completions extends ScalarFunction {
 
+    public static final String P99_METRIC = "com.datasqrl.openai.completions.p99";
```

Comment:
You have the same code here 3 times. Move this into a utility class that takes the function name as the argument and DRYies the code.